doc/tutorials/nvmx_virtualization.rst

```diff
@@ -0,0 +1,314 @@
+* The nested guest VMs, which we call the L2 guest VMs
+
+.. figure:: images/nvmx_1.png
+   :width: 700px
+   :align: center
+
+   Generic Nested Virtualization
+
+
+High Level ACRN Nested Virtualization Design
+********************************************
+
+The high-level design of nested virtualization in ACRN is shown in :numref:`nested_virt_hld`.
+Nested VMX is enabled by allowing a guest VM to use VMX instructions,
+and emulating them using the single level of VMX available in the hardware.
+
+A logical processor uses Virtual-Machine Control Structures (VMCSs) while
+it is in a VMX operation. The core of the NVMX emulation is that ACRN builds VMCS02
+to run the L2 guest VM, out of the VMCS12 built by the L1 hypervisor.
+
+.. figure:: images/nvmx_arch_1.png
+   :width: 700px
+   :align: center
+   :name: nested_virt_hld
+
+   Nested Virtualization in ACRN
+
+
+Restrictions and Constraints
+****************************
+
+Nested virtualization is considered an experimental feature, and only tested
+on the Tiger Lake RVP and Kaby Lake NUC (See :ref:`hardware`.)
+
+L1 VMs have the following restrictions:
+
+* KVM is the only L1 hypervisor supported by ACRN
+* KVM runs in 64-bit mode
+* KVM enables EPT for L2 guests
+* QEMU is used to launch L2 guests
+
+Constraints on L1 guest configuration:
+
+* Local APIC passthrough must be enabled
+* Only SCHED_NOOP scheduler is supported. ACRN can't receive timer interrupts
+  on LAPIC passthrough pCPUs
+* Can not enable UEFI setting ``TCC mode`` on Tiger Lake platform
+
+
+Service OS VM configuration
+***************************
+
+ACRN only supports enabling the nested virtualization feature on the Service VM, not on pre-launched
+VMs.
+
+The nested virtualization feature is disabled by default in ACRN. You can
+enable it using the :ref:`Use the ACRN Configuration Editor <acrn_config_tool_ui>`
+with the settings:
+
+- Configure system level features:
+
+   - Select ``y`` on ``NVMX_ENABLED`` to enable nested virtualization
+
+   - Select ``SCHED_NOOP`` on ``SCHEDULER``
+
+     .. figure:: images/nvmx_cfg_1.png
+        :width: 400px
+        :align: center
+
+        Setting NVMX_ENABLED and SCHEDULER with configuration tool
+
+- In each guest VM configuration:
+
+  - Select ``GUEST_FLAG_NVMX_ENABLED`` on ``guest_flags`` on the SOS VM section
+    to enable the nested virtualization feature on the Service VM.
+  - Select ``GUEST_FLAG_LAPIC_PASSTHROUGH`` on ``guest_flags`` to enable local
+    APIC passthrough on the Service VM.
+
+    .. figure:: images/nvmx_cfg_3.png
+       :width: 700px
+       :align: center
+
+       Service VM (SOS) ``guest_flag`` settings
+
+  - Edit ``cpu_affinity`` to assign ``pCPU`` IDs to run the Service VM. If you are
+    using debug build and need the hypervisor console, don't assign
+    ``pCPU0`` to Service VM.
+
+    You may need to manually edit the ACRN scenario XML configuration file to edit the ``pcpu_id`` for the Service VM (SOS):
```

Comment:
If this is a limitation of the configuration editor, we should fix the editor!  @junjiemao1 @NanlinXie

Reply:
> If this is a limitation of the configuration editor, we should fix the editor! @junjiemao1 @NanlinXie

Without nVMX feature, the CPU affinity assignment of SOS is considered as "all pCPUs minus pCPUs assigned to pre-launched VMs" so it's no need to edit CPU assignment for SOS. Since nested is considered as an experimental feature at this moment and the VM configuration requirement could be changed in the future, I guess we don't have to do this change for now.

Reply:
OK.  In that case, you can delete this section.

Reply:
> OK. In that case, you can delete this section.

But with nVMX enabled, the user may need to edit the CPU affinity for SOS.